Sources/SKTestSupport/String+applyingTextEdits.swift

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+package import LanguageServerProtocol
+
+extension String {
+  /// Returns a new string obtained by applying the given text edits to this
+  /// string.
+  package func applying(_ edits: [TextEdit]) -> String {
+    let sortedEdits = edits.sorted { lhs, rhs in
+      lhs.range.lowerBound < rhs.range.lowerBound
+    }
```

Comment:
I would generally expect this to already be the case, is it not with the new algorithm? We should just be able to go through this list back to front, applying the changes along the way. `RenameAssertions.swift` has a private function that does that already (`apply(edits:to:)`) - we could move that into `SKTestSupport` instead and use it here?